src/modules/battery_status/analog_battery.cpp

```diff
@@ -34,21 +28,25 @@ AnalogBattery::AnalogBattery(int index, ModuleParams *parent) :
 	snprintf(param_name, sizeof(param_name), "BAT%d_A_PER_V", index);
 	_analog_param_handles.a_per_v = param_find(param_name);
 
-	snprintf(param_name, sizeof(param_name), "BAT%d_ADC_CHANNEL", index);
-	_analog_param_handles.adc_channel = param_find(param_name);
+	snprintf(param_name, sizeof(param_name), "BAT%d_V_CHANNEL", index);
+	_analog_param_handles.v_channel = param_find(param_name);
+
+	snprintf(param_name, sizeof(param_name), "BAT%d_I_CHANNEL", index);
+	_analog_param_handles.i_channel = param_find(param_name);
 
 	_analog_param_handles.v_div_old = param_find("BAT_V_DIV");
 	_analog_param_handles.a_per_v_old = param_find("BAT_A_PER_V");
 	_analog_param_handles.adc_channel_old = param_find("BAT_ADC_CHANNEL");
 }
 
 void
-AnalogBattery::updateBatteryStatusRawADC(hrt_abstime timestamp, int32_t voltage_raw, int32_t current_raw,
-		bool selected_source, int priority, float throttle_normalized)
+AnalogBattery::updateBatteryStatusADC(hrt_abstime timestamp, float voltage_raw, float current_raw,
+				      bool selected_source, int priority, float throttle_normalized)
 {
-	float voltage_v = (voltage_raw * _analog_params.cnt_v_volt) * _analog_params.v_div;
-	float current_a = ((current_raw * _analog_params.cnt_v_curr) - _analog_params.v_offs_cur) * _analog_params.a_per_v;
+	float voltage_v = voltage_raw * _analog_params.v_div;
+	float current_a = (current_raw - _analog_params.v_offs_cur) * _analog_params.a_per_v;
 
+	// TODO: voltage_v should be changed to voltage_raw?
```

Comment:
Can we get this resolved? @davids5 ?

Reply:
`voltage_v` is the voltage that v_div applied. But the condition here seems refer to the original ADC sampled voltage.

Reply:
I need more context to answer this, I am lost in the naming.  What is a raw voltage? Is this unfiltered? What is v_div? 

This comparison should be voltage to voltage unfiltered.

Reply:
> I need more context to answer this, I am lost in the naming. What is a raw voltage? Is this unfiltered? What is v_div?
> 
> This comparison should be voltage to voltage unfiltered.

"raw voltage" refers to the value provided by ADC driver directly. If ADC has 3v3 vref, raw voltage would within that range. "v_div" is the short form of "voltage divider", which is the extra resistances that divides battery voltage level down to ADC range. Both of them are unfiltered.

Reply:
Thank you! @SalimTerryLi . So let's delete the comment. In the HW we referrer to the divided voltage as V5_SCALED. This is the effect of the voltage divider. Should we consider renaming the v_div to something line v5_scale_factor?